tx.go

```diff
@@ -652,6 +652,9 @@ func (tx *Tx) Page(id int) (*PageInfo, error) {
 		return nil, nil
 	}
 
+	// Force loading free list if opened in ReadOnly mode.
+	tx.db.loadFreelist()
```

Comment:
Thank you for the PR. 

Everything looks good to me apart this line. As we discussed in #371 , I find it a risk and indeterminism if within a blocking (exclusive lock) transaction we perform lazy initialization with cost of O(db) size. 

I would prefer the call to exit with error that Page(is) is not supported in RO transaction without pre-loaded free pages.

Reply:
Thanks @ptabor for the feedback.

In that case, we need to process https://github.com/etcd-io/bbolt/blob/b654ce922133ff49bb385297f30835ca357bac5f/tx.go#L418-L419 in the same way. But it might be a breaking change.

Reply:
I think it (`db.loadFreelist()`) can stay in check:
   - it's already O(all pages) operation to perform check
   - it's goal it's explicitly to fail in case of any problems is the storage-layer.

Reply:
Done. Please take a look at the third commit https://github.com/etcd-io/bbolt/pull/381/commits/d2fae80500226673ca0fa4bf5a202b3cea97190e 

thx